src/core/mount.c

```diff
@@ -1492,6 +1492,11 @@ static int mount_setup_existing_unit(
                 load_extras = true;
         }
 
+        if (!mount_is_extrinsic(MOUNT(u)) && !u->default_dependencies &&
```

Comment:
I just found that the `!u->default_dependencies` should be inverted, i.e. the ordering dep should only be added if the mount unit has `DefaultDependencies=true`